src/CP/valueselection/classic/impactheuristic.jl

```diff
@@ -0,0 +1,41 @@
+
+"""
+    ImpactHeuristic <: ValueSelection
+ 
+"""
+mutable struct ImpactHeuristic <: ValueSelection
+    search_metrics::Union{Nothing, SearchMetrics}
+end
+
+"""
+    ImpactHeuristic()
+    
+Create the default `ImpactHeuristic` :
+"""
+ImpactHeuristic() = ImpactHeuristic(nothing)
+"""
+    (valueSelection::ImpactHeuristic)(::LearningPhase, model, x, current_status)
+
+Explains what the ImpactHeuristic should do at each step of the solving. This is useful to have a unified `search!` function working with both ImpactHeuristic and LearnedHeuristic. In the case of the ImpactHeuristic, it is only called in the DecisionPhase where the selectValue function is used to choose the value assigned. 
+"""
+(valueSelection::ImpactHeuristic)(::Type{InitializingPhase}, model::Union{Nothing, CPModel}=nothing) = (valueSelection.search_metrics = SearchMetrics(model))
+
+(valueSelection::ImpactHeuristic)(::Type{StepPhase}, model::Union{Nothing, CPModel}=nothing, current_status::Union{Nothing, Symbol}=nothing) = nothing
+
+function (valueSelection::ImpactHeuristic)(::Type{DecisionPhase}, model::Union{Nothing, CPModel}=nothing, x::Union{Nothing, AbstractIntVar}=nothing) 
+    model.statistics.lastVar = x
+    d = Dict()
+    for v in x.domain
+        d[(x,v)] = !isnothing(get(model.impact_var_val, (x,v), nothing)) ? model.impact_var_val[(x,v)] : 0.1
```

Comment:
As the impact value is estimated using past information obtained after assigning a value and running a fix-point, the value is initially unknown, hence we need to set an arbitrary value at the beginning of the search. We set this value to 0.1.